client/app/routes/create-account.tsx

```diff
@@ -9,112 +20,155 @@ import { sendEmailVerification } from "~/lib/send-email-verification";
 import { values } from "~/lib/values.server";
 
 export const loader = async () => {
-	const school = values.get("shortName");
-	const emailExtensions = values.get("emailExtensions") as string[];
+  const school = values.get("shortName");
+  const emailExtensions = values.get("emailExtensions") as string[];
 
-	return { emailExtensions, school };
+  return { emailExtensions, school };
 };
 
 export const action = async ({ request }: ActionFunctionArgs) => {
-	if (request.method !== "POST") {
-		return new Response(null, { status: 405 });
-	}
-
-	const { password, email, username } = await request.json();
-
-	const user = await prisma.user.create({ data: { email, username } });
-	await prisma.authCredential.create({
-		data: { password: await hash(password), userId: user.id },
-	});
-
-	await sendEmailVerification(email);
-
-	return redirect(`/account-created?email=${user.email}`);
+  if (request.method !== "POST") {
+    return new Response(null, { status: 405 });
+  }
+
+  const { password, email, username } = await request.json();
+
+  // check if username already exits
+  const usernameExits = await prisma.user.findUnique({
+    where: {
+      username,
+    },
+  });
+
+  if (usernameExits) {
+    return json(
+      {
+        type: "username-exists",
+        message: "username already exists",
+      },
+      { status: 403 }
+    );
+  }
+
+  // check if email already exits
+  const emailExits = await prisma.user.findUnique({
+    where: {
+      email,
+    },
+  });
+
+  if (emailExits) {
+    return json(
+      { type: "email-exists", message: "email already exits" },
+      { status: 403 }
+    );
+  }
+
+  const user = await prisma.user.create({ data: { email, username } });
```

Comment:
I think you can have a try/catch around this statement and check for conflict error instead of checking one by one. Did you try that?

Reply:
Okay I will try that and resubmit.